More v0.7 updates

alyst opened the pull request:
* drop support for prerelease versions of Julia v0.7
* enable CI testing on 0.7 & 1.0
* update badges
* annotate tests with testsets
* improve support for `iterate()` protocol (define `IteratorSize` and `eltype()`)
* `readstring` was removed from Julia v1.0 (replaced by `read(io, String)`), making the same change in FastaIO would be confusing (FastaIO method is not doing what `read(io, String)` is supposed to do), so I renamed `readstring()` into an exported method `readentries()`

carlobaldassi commented:
Looking good, thanks! I haven't checked all the details yet (will do that in the next few days), but from a cursory look I noticed that the `readstring` -> `readentries` change should get in the docs too (you see this at the bottom of the Travis output too).

Also is there a particular reason for moving the logic of coverage out of `docs/coverage.jl` and into `.travis.yml`?

alyst commented:
>  I noticed that the readstring -> readentries change should get in the docs too 

Sure I can update it. I just don't know whether to keep `readentries()` at all. As you can see, now it's just the synonym of  `collect()` plus there's already `readfasta()` to read all the contents.

> is there a particular reason for moving the logic of coverage out of docs/coverage.jl and into .travis.yml

Initially I was just surprised to discover the coverage script in the docs, and for a minute I was thinking what coverage it can collect there. :) But I also didn't realize that for `Codecov` one really needs these ENV checks (otherwise there would be multiple reports for each job), so the resulting travis.yml line is a little bit long.

I don't have any deep experience with Codecov, but this behaviour looks a bit strange. The [Codecov docs](https://docs.codecov.io/docs/merging-reports) state that by default the reports from multiple platforms should be merged. Maybe you can check whether e.g. `require_ci_to_pass: yes` was disabled in Codecov YAML?

carlobaldassi commented:
Yes, I think we can just remove`readstring`/ `readentries`. (It's possible that `collect` didn't even exist when I first wrote this package...)

As for `Codecov`, I think I had just copied that scheme from somewhere, I don't really remember where from. Your change is fine. I'll have a look at how to merge the reports at some point, but it shouldn't really make a difference for this package, there is nothing platform-specific here, so it's low priority.

alyst commented:
Ok, I removed `readentries()` (also from the docs). For reviewing convenience I've added new commits on top, but prior to merging I can squash the relevant commits together to reduce noise.

carlobaldassi commented:
Thanks a lot!
